srml/system/src/lib.rs

```diff
@@ -14,8 +14,58 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! System manager: Handles lowest level stuff like depositing logs, basic set up and take down of
-//! temporary storage entries, access to old block hashes.
+//! # System module
+//! 
+//! The system module provides low-level access to core types and cross-cutting utilities.
+//! It acts as the base layer for other SRML modules to interact with the Substrate framework components.
+//! To use it in your module, you should ensure your module's trait implies the system [`Trait`].
```

Comment:
Relaying a comment on my PR, we can remove this sentence: "To use it in your module, you should ensure your module's trait implies the system [`Trait`]", since it's redundant in every module

Reply:
I think this more for linking to the module's config trait from the intro. While it is redundant, it still takes the reader/user to the trait documentation.